libs/wingsdk/test/shared-gcp/bucket.inflight.test.ts

```diff
@@ -0,0 +1,320 @@
+import { MockStorage } from "mock-gcs";
+import { vi, test, beforeEach, expect } from "vitest";
+import { BucketClient } from "../../src/shared-gcp/bucket.inflight";
+import { Duration } from "../../src/std";
+
+vi.mock("@google-cloud/storage", () => ({
+  Storage: MockStorage,
+}));
+
+beforeEach(() => {
+  vi.resetAllMocks;
+});
+
+test("put object to bucket", async () => {
+  const BUCKET_NAME = "test-bucket";
+  const KEY = "test-file-1";
+  const VALUE = "test-content-1";
+
+  const storage = new MockStorage();
+
+  const client = new BucketClient(BUCKET_NAME, false, storage as any);
+  const res = await client.put(KEY, VALUE);
+
+  expect(res).toBe(undefined);
```

Comment:
can you test if the right gcp function is called?

Reply:
How about doing it this way? 

For the test `"put object to bucket"`
```js
  const client = new BucketClient(BUCKET_NAME, storage as any);
  await client.put(KEY, VALUE);

  // check that the object was put by ensuring it exists
  const res = await client.exists(KEY);
  expect(res).toBe(true);
  ```
  
  For the test `"delete object from bucket"`
  ```js
  const client = new BucketClient(BUCKET_NAME, storage as any);
  await client.delete(KEY);

  // check that the object was deleted by ensuring it doesn't exist
  const res = await client.exists(KEY);

  expect(res).toBe(false);
  ```

Reply:
yes :)